File: src/main/java/com/zhangziqi/online_course_mine/model/entity/Permission.java
package com.zhangziqi.online_course_mine.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

/**
 * 权限实体类
 */
@Data
@Entity
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "permissions")
@EqualsAndHashCode(callSuper = true)
public class Permission extends BaseEntity {

    /**
     * 权限名称
     */
    @Column(nullable = false, length = 50)
    private String name;

    /**
     * 权限编码
     */
    @Column(nullable = false, unique = true, length = 50)
    private String code;

    /**
     * 权限描述
     */
    @Column(length = 255)
    private String description;

    /**
     * 资源路径
     */
    @Column(length = 255)
    private String url;

    /**
     * HTTP方法
     */
    @Column(length = 10)
    private String method;
} 

File: front/src/types/permission.ts
// 权限实体类型
export interface Permission {
  id: number;
  name: string;         // 权限名称
  code: string;         // 权限编码
  description?: string; // 权限描述
  url?: string;         // 资源URL
  method?: string;      // HTTP方法
  createdAt?: string;   // 创建时间
  updatedAt?: string;   // 更新时间
}

// 权限创建/更新DTO
export interface PermissionDTO {
  id?: number;
  name: string;         // 权限名称
  code: string;         // 权限编码
  description?: string; // 权限描述
  url?: string;         // 资源URL
  method?: string;      // HTTP方法
}

// 分页查询参数
export interface PermissionQueryParams {
  page?: number;
  pageSize?: number;
  name?: string;
  code?: string;
} 